app/javascript/shared/components/ChatForm.vue

```diff
@@ -90,6 +106,32 @@ export default {
       }
       this.$emit('submit', this.formValues);
     },
+
+    validateFormWithApi(name, regex = '') {
+      const formValue = this.formValues[name];
+      if (!regex) {
+        return true;
+      }
+      const [, pattern, flags] = regex.match(/\/(.*)\/([a-z]*)/);
+      const reg = new RegExp(pattern, flags);
+      return reg.test(formValue);
+    },
+
+    isTouched() {
+      this.hasTouched = true;
+    },
+
+    emailValidation(name, type) {
```

Comment:
w can rename the function to be something more generic. The regex doesn't need to be `email`. It could be a specific code, date, anything

Reply:
Changed this to the input pattern attribute method.